sycl/include/sycl/backend_types.hpp

```diff
@@ -21,8 +21,8 @@ enum class backend : char {
   ext_oneapi_level_zero = 2,
   ext_oneapi_cuda = 3,
   all = 4,
-  ext_intel_esimd_emulator __SYCL_DEPRECATED(
-      "esimd emulator is no longer supported") = 5,
+  // No support anymore:
+  // ext_intel_esimd_emulator  = 5,
```

Comment:
I don't know if we should preserve this space in numbering or decrease value for hip/native_cpu by one. Same question for a few other enums.

@sergey-semenov , @steffenlarsen , @AlexeySachkov , WDYT?

Reply:
I guess that values of most (or all) of those enumerations are not user-visible in context that there are no specification which documents exact values. Having gaps in them won't increase size of variables of those enumeration types, because underlying types would still be the same. Therefore, I would vote for leaving the gaps in, simply to reduce amount of things we are breaking.